spec/compiler/codegen/macro_spec.cr

```diff
@@ -1895,4 +1895,34 @@ describe "Code gen: macro" do
       {% end %}
       )).to_i.should eq(10)
   end
+
+  it "returns an array of each call" do
+    run(%(
+      macro test
+        {{@caller.size == 1 ? 1 : 0}}
+      end
+
+      test
+      )).to_i.should eq(1)
+  end
+
+  it "provides access to the macro call stack via @caller" do
+    run(%(
+      macro test(num)
+        {{@caller.last.args[0] == 1 ? 1 : 0}}
```

Comment:
**nitpick:** This should be `first`, not `last`.
If we add more stack frames to `@caller`, they would be added to the end, i.e. the closest one is always the first (same as with `caller`).

```suggestion
        {{@caller.first.args[0] == 1 ? 1 : 0}}
```

Reply:
Ah thanks, was thinking in terms of the trace where the last frame is the one at the end. I'll make a note to include this in the book PR for this ivar.